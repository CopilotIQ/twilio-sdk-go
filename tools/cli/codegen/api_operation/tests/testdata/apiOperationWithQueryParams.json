{
  "packageName": "assistants",
  "name": "Page",
  "imports": ["github.com/RJPearson94/twilio-sdk-go/client"],
  "http": {
    "uri": "/Assistants",
    "method": "Get"
  },
  "options": {
    "name": "AssistantsPageOptions",
    "properties": [
      {
        "name": "DateCreatedAfter",
        "type": "time.Time",
        "value": "DateCreatedAfter",
        "required": false
      },
      {
        "name": "DateCreatedOnOrBefore",
        "type": "time.Time",
        "value": "DateCreatedOnOrBefore",
        "required": false
      },
      {
        "name": "From",
        "type": "string",
        "value": "From",
        "required": false
      },
      {
        "name": "PageSize",
        "type": "int",
        "value": "PageSize",
        "required": false
      },
      {
        "name": "Page",
        "type": "int",
        "value": "Page",
        "required": false
      },
      {
        "name": "PageToken",
        "type": "string",
        "value": "PageToken",
        "required": false
      },
      {
        "name": "To",
        "type": "string",
        "value": "To",
        "required": false
      }
    ]
  },
  "response": {
    "name": "AssistantsPageResponse",
    "type": "JSON",
    "additionalStructs": [
      {
        "name": "PageAssistantResponse",
        "type": "JSON",
        "properties": [
          {
            "name": "AccountSid",
            "type": "string",
            "value": "account_sid",
            "required": true
          },
          {
            "name": "CallbackEvents",
            "type": "string",
            "value": "callback_events",
            "required": false
          },
          {
            "name": "CallbackURL",
            "type": "string",
            "value": "callback_url",
            "required": false
          },
          {
            "name": "DateCreated",
            "type": "time.Time",
            "value": "date_created",
            "required": true
          },
          {
            "name": "DateUpdated",
            "type": "time.Time",
            "value": "date_updated",
            "required": false
          },
          {
            "name": "DevelopmentStage",
            "type": "string",
            "value": "development_stage",
            "required": true
          },
          {
            "name": "FriendlyName",
            "type": "string",
            "value": "friendly_name",
            "required": false
          },
          {
            "name": "LatestModelBuildSid",
            "type": "string",
            "value": "latest_model_build_sid",
            "required": false
          },
          {
            "name": "LogQueries",
            "type": "bool",
            "value": "log_queries",
            "required": true
          },
          {
            "name": "NeedsModelBuild",
            "type": "bool",
            "value": "needs_model_build",
            "required": false
          },
          {
            "name": "Sid",
            "type": "string",
            "value": "sid",
            "required": true
          },
          {
            "name": "UniqueName",
            "type": "string",
            "value": "unique_name",
            "required": true
          },
          {
            "name": "URL",
            "type": "string",
            "value": "url",
            "required": true
          }
        ]
      },
      {
        "name": "PageMetaResponse",
        "type": "JSON",
        "properties": [
          {
            "name": "FirstPageURL",
            "type": "string",
            "value": "first_page_url",
            "required": true
          },
          {
            "name": "Key",
            "type": "string",
            "value": "key",
            "required": true
          },
          {
            "name": "NextPageURL",
            "type": "string",
            "value": "next_page_url",
            "required": false
          },
          {
            "name": "Page",
            "type": "int",
            "value": "page",
            "required": true
          },
          {
            "name": "PageSize",
            "type": "int",
            "value": "page_size",
            "required": true
          },

          {
            "name": "PreviousPageURL",
            "type": "string",
            "value": "previous_page_url",
            "required": false
          },
          {
            "name": "URL",
            "type": "string",
            "value": "url",
            "required": true
          }
        ]
      }
    ],
    "properties": [
      {
        "name": "Assistants",
        "type": "[]PageAssistantResponse",
        "value": "assistants",
        "required": true
      },
      {
        "name": "Meta",
        "type": "PageMetaResponse",
        "value": "meta",
        "required": true
      }
    ]
  }
}
