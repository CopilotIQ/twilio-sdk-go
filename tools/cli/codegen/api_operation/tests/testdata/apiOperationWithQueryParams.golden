// This is an autogenerated file. DO NOT MODIFY
package assistants

import (
	"context"
	"net/http"
	"time"

	"github.com/RJPearson94/twilio-sdk-go/client"
	"github.com/RJPearson94/twilio-sdk-go/utils"
)

type AssistantsPageOptions struct {
	DateCreatedAfter      *time.Time
	DateCreatedOnOrBefore *time.Time
	From                  *string
	PageSize              *int
	Page                  *int
	PageToken             *string
	To                    *string
}

type PageAssistantResponse struct {
	AccountSid          string     `json:"account_sid"`
	CallbackEvents      *string    `json:"callback_events,omitempty"`
	CallbackURL         *string    `json:"callback_url,omitempty"`
	DateCreated         time.Time  `json:"date_created"`
	DateUpdated         *time.Time `json:"date_updated,omitempty"`
	DevelopmentStage    string     `json:"development_stage"`
	FriendlyName        *string    `json:"friendly_name,omitempty"`
	LatestModelBuildSid *string    `json:"latest_model_build_sid,omitempty"`
	LogQueries          bool       `json:"log_queries"`
	NeedsModelBuild     *bool      `json:"needs_model_build,omitempty"`
	Sid                 string     `json:"sid"`
	UniqueName          string     `json:"unique_name"`
	URL                 string     `json:"url"`
}

type PageMetaResponse struct {
	FirstPageURL    string  `json:"first_page_url"`
	Key             string  `json:"key"`
	NextPageURL     *string `json:"next_page_url,omitempty"`
	Page            int     `json:"page"`
	PageSize        int     `json:"page_size"`
	PreviousPageURL *string `json:"previous_page_url,omitempty"`
	URL             string  `json:"url"`
}

type AssistantsPageResponse struct {
	Assistants []PageAssistantResponse `json:"assistants"`
	Meta       PageMetaResponse        `json:"meta"`
}

func (c Client) Page(options *AssistantsPageOptions) (*AssistantsPageResponse, error) {
	return c.PageWithContext(context.Background(), options)
}

func (c Client) PageWithContext(context context.Context, options *AssistantsPageOptions) (*AssistantsPageResponse, error) {
	op := client.Operation{
		Method:      http.MethodGet,
		URI:         "/Assistants",
		QueryParams: utils.StructToStringMap(options),
	}

	response := &AssistantsPageResponse{}
	if err := c.client.Send(context, op, nil, response); err != nil {
		return nil, err
	}
	return response, nil
}
